src/main/java/com/cloudbees/jenkins/support/impl/CustomLogs.java

```diff
@@ -95,6 +95,30 @@ public Iterable<LogRecord> getLogRecords() {
                     }
                 });
             }
+
+            // Add rotated log files
+            File[] rotatedLogFiles = customLogs.listFiles((dir, filename) -> filename.matches(name + "\\.log\\.\\d+"));
+            if (rotatedLogFiles == null) {
+                LOGGER.fine("No rotated logs found for : " + name);
+                return;
+            }
+
+            for (File rotatedLogFile : rotatedLogFiles) {
+                String rotatedEntryName = "nodes/master/logs/custom/{0}.log.{1}";
+
+                try {
+                    // Fetch the rotations number of the log
+                    // eg. custom.log.2 , the rotation number is 2
+                    String[] logNameParts = rotatedLogFile.getName().split("\\.");
+                    String logRotationNumber = logNameParts[logNameParts.length - 1];
+
+                    result.add(
+                            new FileContent(rotatedEntryName, new String[] {name, logRotationNumber}, rotatedLogFile));
+                } catch (Exception e) {
+                    LOGGER.warning(String.format(
+                            "Error while adding rotated log files for '%s'. Error: %s", name, e.getMessage()));
```

Comment:
```suggestion
                    LOGGER.log(Level.WARNING, "Error while adding rotated log files for " + name, e);
```
so as not to swallow stack traces